Classes/Controller/AbstractController.php

```diff
@@ -108,28 +126,33 @@ protected function initialize()
      *
      * @access protected
      *
-     * @param array $requestData: The request data
+     * @param  int $documentId: The document's UID (fallback: $this->requestData[id])
      *
      * @return void
      */
-    protected function loadDocument($requestData)
+    protected function loadDocument(int $documentId = 0)
     {
+        // Get document ID from request data if not passed as parameter.
+        if ($documentId === 0 && !empty($this->requestData['id'])) {
+            $documentId = $this->requestData['id'];
+        }
+
         // Try to get document format from database
-        if (!empty($requestData['id'])) {
+        if (!empty($documentId)) {
 
             $doc = null;
 
-            if (MathUtility::canBeInterpretedAsInteger($requestData['id'])) {
+            if (MathUtility::canBeInterpretedAsInteger($documentId)) {
                 // find document from repository by uid
-                $this->document = $this->documentRepository->findOneByIdAndSettings((int) $requestData['id']);
+                $this->document = $this->documentRepository->findOneByIdAndSettings((int) $documentId);
```

Comment:
Do you still need parse it here to int?

In method paramater it is already passed as int.

```suggestion
                $this->document = $this->documentRepository->findOneByIdAndSettings($documentId);
```